p2panda-sync/src/log_sync.rs

```diff
@@ -50,18 +49,9 @@ pub struct LogSyncProtocol<TM, L, E> {
 impl<'a, T, TM, L, E> SyncProtocol<T, 'a> for LogSyncProtocol<TM, L, E>
 where
     T: Topic,
-    TM: Debug + TopicMap<T, Logs<L>> + Send + Sync,
-    L: Clone
-        + Debug
-        + Default
-        + Eq
-        + StdHash
-        + Send
-        + Sync
-        + for<'de> Deserialize<'de>
-        + Serialize
-        + 'a,
-    E: Clone + Debug + Default + Send + Sync + for<'de> Deserialize<'de> + Serialize + 'a,
+    TM: TopicMap<T, Logs<L>>,
+    L: LogId + for<'de> Deserialize<'de> + Serialize + 'a,
```

Comment:
We run into some complexity if we add the `for<'de> Deserialize<'de>` bound directly to the `LogId` trait (specifically, an "implementation of `Deserialize` is not general enough" error). I imagine there's a work-around but it'll be quite technical and I don't want to pour time into understanding the solution right now.

`Serialize` could be included as an implicit bound but I think it's clearer if `Deserialize` and `Serialize` are both explicitly added as bounds.

This applies to `LogId` and `Extensions`.

Reply:
Oh dammit! How did we make it work for `Topic`? There we have the serde traits defined, I wonder what makes it problematic, is it the defaults?

Reply:
So, thankfully this issue does not actually apply to `Extensions`; it works fine for us to include the `Deserialize` and `Serialize` bounds there.

It's just an issue for `LogId`. If we add the bounds to the supertrait, the memory store breaks but log sync is fine. If we remove the bounds then we have to include them explicitly for log sync. This is just in a single place so maybe it's not a terrible outcome?

Reply:
Makes sense! Also, `LogId` is quite a special thing, the core data types do not require it, but it's used as soon as data is persisted or synced, for the latter we need serde as it is sent over the wire (at least for log-height sync, maybe it's not required for set reconciliation).

Reply:
Yeh, I agree that `LogId` is a bit different from `Extensions`, it's not an inherently "on the wire" data type, whereas `Extensions` are. So having to add serde traits where needed maybe isn't so bad.